File: python/507.py
class Solution:
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 1:
            return False
        sums = 1
        factor = 2
        while factor <= int(num ** 0.5):
            if num % factor == 0:
                sums += factor
                sums += num//factor
            factor += 1
        return sums == num

File: go/507.go
import "math"

func checkPerfectNumber(num int) bool {
	if num < 4 {
		return false
	}
	sum := 0
	for i := 1; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			sum += i + num/i
		}
	}
	return sum == 2*num
}